Microsoft.Toolkit.Uwp.UI.Controls/Carousel/Carousel.cs

```diff
@@ -182,7 +182,7 @@ private static void OnCarouselPropertyChanged(DependencyObject d, DependencyProp
 
             Carousel carouselControl = (Carousel)d;
 
-            if (e.Property == SelectedIndexProperty && carouselControl.Items.Count > (int)e.NewValue)
+            if (e.Property == SelectedIndexProperty && (int)e.NewValue >= 0 && carouselControl.Items.Count > (int)e.NewValue)
```

Comment:
`(int)e.NewValue >= 0` will introduce one more unboxing operation. To avoid extra unboxing in the new changes only `carouselControl.Items.Count >= 0` check can be used. But I'd prefer to unbox once and use unboxed value everywhere within this method. I mean
```
if (e.NewValue == null)
{
    return;
}

int newValue = (int)newValue;
Carousel carouselControl = (Carousel)d;

if (e.Property == SelectedIndexProperty && newValue >= 0 && carouselControl.Items.Count > newValue)
{
    var item = carouselControl.Items[newValue];
    carouselControl.FocusContainerFromIndex(newValue);
...
```